### PR TITLE
Add support for ECR gate to MPS

### LINKED ISSUE #2118: Kernel dies when ECR in basis gate set for MPS
### Informations

- **Qiskit Aer version**: 0.14.1
- **Python version**: 3.11.8
- **Operating system**: MacOS

### What is the current behavior?

Kernel dies when using ECR basis gate in MPS simulator. 

### Steps to reproduce the problem

```python
from qiskit import QuantumCircuit, transpile
from qiskit_aer import AerSimulator

qc = QuantumCircuit(2)
qc.h(0)
qc.cx(0, 1)
qc.rz(0.5, 0)
qc.measure_all()

# Works with CX
simulator = AerSimulator(method="matrix_product_state", basis_gates=["cx", "id", "rz", "sx"])
qct = transpile(qc, simulator)
display(qct.draw(output='mpl'))
job = simulator.run(qct)

# Kernel dies with ECR
simulator = AerSimulator(method="matrix_product_state", basis_gates=["ecr", "id", "rz", "sx"])
qct = transpile(qc, simulator)
display(qct.draw(output='mpl'))
job = simulator.run(qct)
```

Last line causes Kernel to die every time. 

### What is the expected behavior?
 Kernel should not die, simulation should be able to use CX or ECR basis gate, esp. as new devices all use ECR. 

### Suggested solutions
Unsure

### PR DESCRIPTION
**Pull request by doichanj**
### Summary

Fixes #2118 

### Details and comments

- Add native support of ECR gate in MPS and extended_stabilizer methods
- (Also added for tensor_network method that was supported but not in basis_gates)
- Add check of basis_gates passed to backend options if there is unsupported gate for the method
- Fix test case for ECR gate

### CODE REVIEW

File: src/simulators/extended_stabilizer/gates.hpp [hunk 1]
```diff
@@ -85,6 +86,7 @@ const AER::stringmap_t<Gatetypes> gate_types_ = {
     {"cx", Gatetypes::clifford},   // Controlled-X gate (CNOT)
     {"cz", Gatetypes::clifford},   // Controlled-Z gate
     {"swap", Gatetypes::clifford}, // SWAP gate
+    {"ecr", Gatetypes::clifford},  // ECR Gate
```
Comment (enum-class) [inline src/simulators/extended_stabilizer/gates.hpp:89]: Hi, I am a junior in the realm of Quantum, Just out of curiosity, Is ECR gate a clifford gate ?

File: src/simulators/extended_stabilizer/extended_stabilizer_state.hpp [hunk 2]
```diff
@@ -694,6 +695,16 @@ void State::apply_gate(const Operations::Op &op, RngEngine &rng, uint_t rank) {
...
     break;
+  case Gates::ecr:
+    BaseState::qreg_.apply_h(op.qubits[1], rank);
+    BaseState::qreg_.apply_s(op.qubits[0], rank);
+    BaseState::qreg_.apply_z(op.qubits[1], rank); // sdg(1)
+    BaseState::qreg_.apply_s(op.qubits[1], rank); // sdg(1)
+    BaseState::qreg_.apply_h(op.qubits[1], rank);
+    BaseState::qreg_.apply_cx(op.qubits[0], op.qubits[1], rank);
+    BaseState::qreg_.apply_x(op.qubits[0], rank);
+    BaseState::qreg_.apply_x(op.qubits[1], rank);
```
Comment (enum-class) [inline src/simulators/extended_stabilizer/extended_stabilizer_state.hpp:706]: This implementation differs to ecr by a -pi/4 global phase. is it generally ok ?

Comment (enum-class) [reply src/simulators/extended_stabilizer/extended_stabilizer_state.hpp:706]: If I am not mistaken this one doesn't have phase difference 

qc.h(1)
qc.cx(0, 1)
qc.t(1)
qc.cx(0, 1)
qc.h(1)
qc.x(0)
qc.h(1)
qc.cx(0, 1)
qc.tdg(1)
qc.cx(0, 1)
qc.h(1)

Comment (doichanj) [reply src/simulators/extended_stabilizer/extended_stabilizer_state.hpp:706]: I coped from stabilizer

https://github.com/Qiskit/qiskit-aer/blob/b706ee1c7445f13f11a28eff4e11a006c89a78dd/src/simulators/stabilizer/stabilizer_state.hpp#L389-L397

But it seems it was not correct, ad I get correct circuit by using transpiler

```
from qiskit import transpile
from qiskit import QuantumCircuit
from qiskit_aer import AerSimulator
sim = AerSimulator(basis_gates = ["h", "cx", "x", "s", "z"])
qc = QuantumCircuit(2)
qc.ecr(0, 1)
qc = transpile(qc,backend=sim,optimization_level=0)
qc.draw('mpl')
```
image

I will fix by using this circuit

**Review by hhorii — APPROVED** [review]

**Review by hhorii — APPROVED** [review]

**Comment by zlatko-minev** [comment]
Great thank you

**Pull request merged**